### PR TITLE
cdc: return server_is_busy to cdc clients if necessary

### PR DESCRIPTION
**Pull request by hicqu**
### What is changed and how it works?

Issue Number: Ref #16035 

What's Changed:

```commit-message
return server_is_busy to cdc clients if necessary
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/cdc/src/endpoint.rs [hunk 1]
```diff
@@ -742,10 +742,7 @@ impl<T: 'static + CdcHandle<E>, E: KvEngine, S: StoreRegionMeta> Endpoint<T, E,
             );
             // To avoid OOM (e.g., https://github.com/tikv/tikv/issues/16035),
             // TiKV needs to reject and return error immediately.
-            //
-            // TODO: TiKV is supposed to return a "busy" error, but for the sake
-            // of compatibility, it returns a "region not found" error.
-            let _ = downstream.sink_region_not_found(region_id);
+            let _ = downstream.sink_server_is_busy(region_id);
```
Comment (overvenus) [inline components/cdc/src/endpoint.rs:745]:
```suggestion
            let _ = downstream.sink_server_is_busy(region_id, "too many pending incremental scans".to_owned());
```

**Review by overvenus — COMMENTED** [review]
Please verify if the previous version of TiCDC can handle the new busy error properly.
Rest LGTM

**Review by overvenus — APPROVED** [review]

File: components/cdc/src/endpoint.rs [hunk 2]
```diff
@@ -742,10 +742,8 @@ impl<T: 'static + CdcHandle<E>, E: KvEngine, S: StoreRegionMeta> Endpoint<T, E,
             );
             // To avoid OOM (e.g., https://github.com/tikv/tikv/issues/16035),
             // TiKV needs to reject and return error immediately.
-            //
-            // TODO: TiKV is supposed to return a "busy" error, but for the sake
-            // of compatibility, it returns a "region not found" error.
-            let _ = downstream.sink_region_not_found(region_id);
+            let _ = downstream
+                .sink_server_is_busy(region_id, "too many pending incremental scans".to_owned());
```
Comment (Connor1996) [inline components/cdc/src/endpoint.rs:746]: add it to the server is busy metric?

**Review by Connor1996 — APPROVED** [review]
LGTM

File: components/cdc/src/endpoint.rs (hunk 2 above)
Comment (hicqu) [reply components/cdc/src/endpoint.rs:746]: TiCDC requests are not important as TiDB requests. So seems it's better to seperate their metrics and alerts.

**Pull request merged**